src/test/javascript/spec/component/admin/metrics-modal-threads.component.spec.ts

```diff
@@ -0,0 +1,168 @@
+            expect(comp.getBgClass(ThreadState.Waiting)).toEqual('bg-info');
+            expect(comp.getBgClass(ThreadState.TimedWaiting)).toEqual('bg-warning');
+            expect(comp.getBgClass(ThreadState.Blocked)).toEqual('bg-danger');
+            expect(comp.getBgClass(ThreadState.New)).toEqual('');
+            expect(comp.getBgClass(ThreadState.Terminated)).toEqual('');
+        });
+    });
+
+    describe('filters', () => {
+        describe('threads based on selected thread state', () => {
+            it('when selected thread state is undefined, nothing is filtered out', () => {
+                // GIVEN
+                comp.threads = runnableThreads.concat(waitingThreads);
+                comp.selectedThreadState = undefined;
+
+                // WHEN
+                comp.refreshTheFilter();
+
+                // THEN
+                expect(comp.filteredThreads).toEqual(runnableThreads.concat(waitingThreads));
+            });
+
+            it('when selected a specific thread state, only threads with specific thread state are returned', () => {
+                // GIVEN
+                comp.threads = runnableThreads.concat(waitingThreads);
+                comp.selectedThreadState = ThreadState.Runnable;
+
+                // WHEN
+                comp.refreshTheFilter();
+
+                // THEN
+                expect(comp.filteredThreads).toEqual(runnableThreads);
+            });
+        });
+
+        describe('threads based on filter text', () => {
+            it('when filter is undefined, nothing is filtered out', () => {
+                // GIVEN
+                comp.threads = runnableThreads.concat(waitingThreads);
+                comp.threadFilter = undefined;
+
+                // WHEN
+                comp.refreshTheFilter();
+
+                // THEN
+                expect(comp.filteredThreads).toEqual(runnableThreads.concat(waitingThreads));
+            });
+
+            it('when filter is entered, only matching results are returned', () => {
+                // GIVEN
+                comp.threads = runnableThreads.concat(waitingThreads);
+                comp.threadFilter = '2';
+
+                // WHEN
+                comp.refreshTheFilter();
+
+                // THEN
+                expect(comp.filteredThreads).toEqual([runnableThreads[1]]);
+            });
+        });
+
+        it('both on thread state and filter text', () => {
+            // GIVEN
+            comp.threads = runnableThreads.concat(waitingThreads);
+            comp.threadFilter = '2';
+            comp.selectedThreadState = ThreadState.Waiting;
+
+            // WHEN
+            comp.refreshTheFilter();
+
+            // THEN
+            expect(comp.filteredThreads).toEqual([]);
+        });
+    });
+
+    describe('on dismiss', () => {
+        it('calls activeModal.dismiss()', () => {
+            // GIVEN
+            let dismissSpy = jest.spyOn(activeModal, 'dismiss').mockImplementation(() => {});
```

Comment:
just a reminder for client-style issue 👼 since you are using spyOn do we need jest.clearAllMocks() / mockClear() in here or maybe in afterEach block? or its just one spy, is not clearing/restoring needed for this case? I'm just trying to learn, sorry :/